In [22]:
import pandas as pd

In [23]:
# March 2024 Green Taxi data
url = 'https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2024-03.parquet'
df = pd.read_parquet(url)

In [24]:
len(df)

57457

Q2. Metric
Let's expand the number of data quality metrics we’d like to monitor!
Please add one metric of your choice and a quantile value for the "fare_amount" column (quantile=0.5).

Hint: explore evidently metric ColumnQuantileMetric (from evidently.metrics import ColumnQuantileMetric)

Q: What metric did you choose?

A: df.quantile

In [25]:
quantile_50 = df['fare_amount'].quantile(0.5)
quantile_50

np.float64(13.5)

Q3. Monitoring
Let’s start monitoring. Run expanded monitoring for a new batch of data (March 2024).

Q: What is the maximum value of metric quantile = 0.5 on the "fare_amount" column during March 2024 (calculated daily)?

A: 13.5

Q4. Dashboard
Finally, let’s add panels with new added metrics to the dashboard. After we customize the dashboard let's save a dashboard config, so that we can access it later. Hint: click on “Save dashboard” to access JSON configuration of the dashboard. This configuration should be saved locally.

Q: Where to place a dashboard config file?

A: (05-monitoring/config)